From: Jordan Justen <jordan.l.justen@intel.com>
Date: Fri, 24 Jul 2020 15:36:34 -0700
Subject: renderdoc/serialise: Add DoStringise(const long &el) for 32-bit x86
 on GCC

linux_process.cpp was generating linker errors:

undefined reference to `rdcstr DoStringise<long>(long const&)

Signed-off-by: Jordan Justen <jordan.l.justen@intel.com>
---
 renderdoc/serialise/serialiser.cpp | 8 ++++++++
 1 file changed, 8 insertions(+)

--- a/renderdoc/serialise/serialiser.cpp
+++ b/renderdoc/serialise/serialiser.cpp
@@ -985,6 +985,14 @@ rdcstr DoStringise(const int16_t &el)
   return StringFormat::Fmt("%hd", el);
 }
 
+#if ENABLED(RDOC_LINUX) && ENABLED(RDOC_X86_FAMILY) && DISABLED(RDOC_X64)
+template <>
+rdcstr DoStringise(const long &el)
+{
+  return StringFormat::Fmt("%l", el);
+}
+#endif
+
 template <>
 rdcstr DoStringise(const float &el)
 {
